haskell/test/Test/FA2/Transfer.hs

```diff
@@ -0,0 +1,160 @@
+test_zero_transfers =
+  nettestScenarioOnEmulatorCaps "transfer is always accepted when the amount is 0" do
+    owner <- newAddress auto
+    operator <- newAddress auto
+    cfmm <- originateSegCFMMWithBalances [owner]
+
+    -- the token does not even exist
+    withSender owner $ transferToken cfmm owner operator (FA2.TokenId 0) 0
+
+    setSimplePosition cfmm owner -10 15
+    withSender owner $ transferToken cfmm owner operator (FA2.TokenId 0) 0
+
+    setSimplePosition cfmm owner -20 -15
+    withSender owner $ updateOperator cfmm owner operator (FA2.TokenId 1) True
+    withSender operator $ transferToken cfmm owner operator (FA2.TokenId 1) 0
+
+test_unknown_position :: TestTree
+test_unknown_position =
+  nettestScenarioOnEmulatorCaps "transfer does not accept unknown positions" do
+    owner <- newAddress auto
+    receiver <- newAddress auto
+    cfmm <- originateSegCFMMWithBalances [owner]
+
+    expectCustomError_ #fA2_TOKEN_UNDEFINED $
+      withSender owner $ transferToken' cfmm owner receiver (FA2.TokenId 0)
+
+test_not_owner :: TestTree
+test_not_owner =
+  nettestScenarioOnEmulatorCaps "transfer rejects non-owned/operated positions" do
+    owner <- newAddress auto
+    notOwner <- newAddress auto
+    receiver <- newAddress auto
+    cfmm <- originateSegCFMMWithBalances [owner, notOwner, receiver]
+
+    setSimplePosition cfmm owner -10 15
+    expectCustomError #fA2_INSUFFICIENT_BALANCE (#required .! 1, #present .! 0) $
+      withSender notOwner $ transferToken' cfmm notOwner receiver (FA2.TokenId 0)
+
+test_not_operator :: TestTree
+test_not_operator =
+  nettestScenarioOnEmulatorCaps "transfer rejects invalid operators" do
+    owner <- newAddress auto
+    notOper <- newAddress auto
+    cfmm <- originateSegCFMMWithBalances [owner, notOper]
+
+    setSimplePosition cfmm owner -10 15
+    expectCustomError_ #fA2_NOT_OPERATOR $
+      withSender notOper $ transferToken' cfmm owner notOper (FA2.TokenId 0)
+
+test_fungible_amount :: TestTree
+test_fungible_amount =
+  nettestScenarioOnEmulatorCaps "transfer rejects amounts higher than 1" do
+    owner <- newAddress auto
+    receiver <- newAddress auto
+    cfmm <- originateSegCFMMWithBalances [owner]
+
+    setSimplePosition cfmm owner -10 15
+    expectCustomError #fA2_INSUFFICIENT_BALANCE (#required .! 2, #present .! 1) $
+      withSender owner $ transferToken cfmm owner receiver (FA2.TokenId 0) 2
+
+test_owner_transfer :: TestTree
+test_owner_transfer =
+  nettestScenarioOnEmulatorCaps "transfer moves positions when called by owner" do
+    owner <- newAddress auto
+    receiver <- newAddress auto
+    cfmm <- originateSegCFMMWithBalances [owner]
+
+    setSimplePosition cfmm owner -10 15
+    balanceOf cfmm (FA2.TokenId 0) owner @@== 1
+    balanceOf cfmm (FA2.TokenId 0) receiver @@== 0
+    withSender owner $ transferToken' cfmm owner receiver (FA2.TokenId 0)
+    balanceOf cfmm (FA2.TokenId 0) owner @@== 0
+    balanceOf cfmm (FA2.TokenId 0) receiver @@== 1
+
```

Comment:
Minor: I would also check that the new owner can manage the position, and the old owner can't.